packages/wordcount/src/index.js

```diff
@@ -95,10 +95,12 @@ export function count( text, type, userSettings ) {
 	const settings = loadSettings( type, userSettings );
 	if ( text ) {
 		let matchRegExp = settings[ type + 'RegExp' ];
-		if ( 'words' === settings.type ) {
-			return matchWords( text, matchRegExp, settings ).length;
-		} else {
-			return matchCharacters( text, matchRegExp, settings ).length;
+		const results = ( 'words' === settings.type ) ?
+			matchWords( text, matchRegExp, settings ) :
+			matchCharacters( text, matchRegExp, settings );
+		if ( results ) {
+			return results.length;
```

Comment:
And I just realised that if we want to go full ternary here you could do:

```js
return results ? results.length : 0;
```

In this case I don't think the ternary hurts readability, so if you want to feel free to do that. 🤷‍♂️